shuup/core/models/_categories.py

```diff
@@ -129,6 +130,30 @@ def _get_slug_name(self):
             return None
         return self.safe_translation_getter("name")
 
+    def delete(self, using=None):
+        raise NotImplementedError("Not implemented: Use `soft_delete()` for categories.")
+
+    @atomic
+    def soft_delete(self, user=None):
+        if not self.status == CategoryStatus.DELETED:
+            for shop_product in self.primary_shop_products.all():
+                shop_product.primary_category = None
+                shop_product.categories.remove(self)
```

Comment:
Unnecesary remove since the category is removed in the next phase from shop_product categories.